docs/src/main/asciidoc/init-tasks.adoc

```diff
@@ -0,0 +1,145 @@
+configuration prefix, but all the configuration options are also available for OpenShift(`quarkus.openshift` prefix) too.
+
+In the case where we need to disable a particular task, we can use the following property:
+
+[source,properties]
+----
+quarkus.kubernetes.init-tasks."<task name>".enabled=false
+----
+
+The task name is the name of the extension that performs the initialization.
+Examples:
+
+For Flyway:
+
+[source,properties]
+----
+quarkus.kubernetes.init-tasks.flyway.enabled=false
+----
+
+For Liquibase:
+
+[source,properties]
+----
+quarkus.kubernets.init-tasks.liquibase.enabled=false
+----
+
+For Liquibase Mongodb:
+
+[source,properties]
+----
+quarkus.kubernetes.init-tasks.liquibase-mongodb.enabled=false
+----
+
+
+== Controlling the generated job
+
+The job container is pretty similar to the application container, and the only thing that changes is the configured environment variables.
+More specifically, the following environment variable is added, to tell the job to exit right after initialization.
+
+[source,properties]
+----
+QUARKUS_INIT_AND_EXIT=true
+----
+
+The image, image pull policy, service account, volumes, mounts and additional environment variables are inherited/copied from the deployment resource.
+Any customization to the original deployment resource (via configuration or extension) will also be reflected in the job too.
+
+== Controlling the generated init container
+
+The name of the generated init container is `wait-for-${task name}` by default.
+Given that the init container is part of the same pod as the actual application it will get the same service account (and therefore permissions) and volumes as the application.
+Further customization to the container can be done using using the configuration options for init containers (see `quarkus.kubernetes.init-containers` or `quarkus.openshift.init-containers`).
+
+Examples:
+
+To set the imagePullPolicy to `IfNotPresent` on the init container that waits for the `flyway` job:
+
+[source,properties]
+----
+quarkus.kubernetes.init-containers.wait-for-flyway.image-pull-policy=IfNotPresent
+----
+
+To set custom command (say `custom-wait-for`) on the init container that waits for the `flyway` job:
+
+[source,properties]
+----
+quarkus.kubernetes.init-containers.wait-for-flyway.command=custom-wait-for
+----
+
+
+== Orchestration of the initialization tasks
+
+The deployment resource should not start until the job has been completed. The typical pattern that is used among Kubernetes users is the
+use of init containers to achieve this. An init container that `wait for` the job to complete is enough to enforce that requirement.
+
+=== Using a custom wait-for container image
+
+To change the `wait-for` image which by default is `groundnuty/k8s-wait-for:no-root-v1.7` you can use:
+
+[source,properties]
+----
+quarkus.kubernetes.init-task-defaults.wait-for-image=my/wait-for-image:1.0
```

Comment:
not sure if this is just how it was done or new names but feels we are being somewhat inconsistent in naming here:

`quarkus.kubernetes.init-task-defaults.wait-for-image` vs `quarkus.kubernetes.init-containers.wait-for-flyway`

should it be `init-task-defaults` and `-init-task.wait-for-flyway` or `init-containers-defaults` and `init-containers.wait-for-flyway` ? 

and should `wait-for-flyway` be `wait-for-flyway-image` ?

Reply:
I agree with @maxandersen, it looks more consistent.

Reply:
> and should `wait-for-flyway` be `wait-for-flyway-image` ?

In `quarkus.kubernetes.init-container.wait-for-flyway.image` the `wait-for-flyway` part is the container name. This is part of the container customization options that we have since forever.

The `quarkus.kubernetes.init-task-defaults` is a new object I added to hold a couple of flags and a default image. 

I could possibly change the way we pass the image to be more aligned with how we configure containers and make it something like `quarkus.kubernetes.init-task-defaults.wait-for-container.image`. Would that work better @maxandersen @cescoffier ?

Reply:
+1